modules/update-cvo-conditions.adoc

```diff
@@ -0,0 +1,65 @@
+
+NoDesiredImage:: The CVO has not been given a release image to reconcile. When NoDesiredImage is True, it is a CVO coding error. The `desiredUpdate` field returns you to the current CVO's release image.
+
+ClusterOperatorNotAvailable:: `ClusterOperatorNotAvailable` is set when the CVO fails to retrieve the ClusterOperator from the cluster or when the retrieved ClusterOperator does not satisfy the reconciliation conditions.
++
+Unlike most manifest-reconciliation failures, this error does not immediately result in `Failing=True`. Under some conditions during installs and updates, the CVO will treat this condition as a `Progressing=True` condition and give the operator up to forty minutes to level before reporting `Failing=True`.
+
+RetrievedUpdates:: When `RetrievedUpdates` is set to True, the CVO successfully retrieves updates. When the `RetrievedUpdates` is set to False, the cluster is not able to retrieve recommended updates. When a CVO is unable to retrieve recommended updates the `CannotRetrieveUpdates` alert displays the reason for not retrieving recommended updates.
+
+// When `RetrievedUpdates` is False, reason will be set to explain why, as discussed in the following subsections.
+// In all cases, the impact is that the cluster will not be able to retrieve recommended updates, so cluster admins will need to monitor for available updates on their own or risk falling behind on security or other bugfixes.
+// When CVO is unable to retrieve recommended updates the CannotRetrieveUpdates alert will fire containing the reason.
+// This alert will not fire when the reason updates cannot be retrieved is NoChannel.
+
+NoUpstream:: No upstream server has been set to retrieve updates.
++
+Set `spec.upstream` in `ClusterVersion` to point to a link:https://github.com/openshift/cincinnati/blob/master/docs/design/openshift.md[Cincinnati server] to receive updates.
+
+InvalidURI:: The configured upstream URI is not valid.
++
+Configure the upstream URI by setting `spec.upstream` in `ClusterVersion` to point to a valid link:https://github.com/openshift/cincinnati/blob/master/docs/design/openshift.md[Cincinnati URI].
+
+InvalidID:: The configured clusterID is not a valid UUID.
++
+To configure the clusterID, set `spec.clusterID` to a valid UUID. The UUID should be unique for each cluster. The default value is used report Telemetry and Insights. It may also be used by the CVO when making Cincinnati requests, so that Cincinnati can return update recommendations tailored to the specific cluster.
+
+NoArchitecture:: The set of architectures has not been configured. This is a result of a CVO coding error. There is no mitigation except to upgrade to a new release image with a fixed CVO.
+
+Impact:: The cluster is unable to retrieve recommended updates. This is a result of a CVO coding error. There is no mitigation except to upgrade to a new release image with a fixed CVO.
```

Comment:
[This is a subsection of the `NoArchitecture` section][1].  But the CVO-bug side of this is already covered in the above `NoArchitecture` wording, and the generic `RetrievedUpdates=False` impact is commented out in the section about the condition type, and probably doesn't need to be repeated under each of the underlying reasons.

[1]: https://github.com/openshift/cluster-version-operator/blame/c07975c452303469c378bc5b0f70d8763eaca1f3/docs/user/status.md#L54-L66